sdk/textanalytics/ai-text-analytics/src/cognitiveServicesCredential.ts

```diff
@@ -1,24 +1,53 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-import { ApiKeyCredentials } from "@azure/core-http";
+import { ServiceClientCredentials, WebResource } from "@azure/core-http";
+
+const SUBSCRIPTION_KEY_HEADER_NAME = "Ocp-Apim-Subscription-Key";
 
 /**
- * Creates a new CognitiveServicesCredential object.
- *
- * @param {string} subscriptionKey   The CognitiveServices subscription key
+ * This class is used to authenticate to Cognitive Services using a Subscription Key retrieved
+ * from the Azure portal.
  */
-export class CognitiveServicesCredential extends ApiKeyCredentials {
+export class CognitiveServicesCredential implements ServiceClientCredentials {
```

Comment:
Per other comment, in other languages we'll keep this TextAnalytics-specific since we haven't seen enough of the cognitive services yet to generalize to all of them.

Reply:
I'm fine doing the name change, but my understanding is that all of the cognitive services today use the same header to authenticate with an opaque string token?

Reply:
That is to say, we already provide this class for all Track 1 libraries

Reply:
I renamed the type for now